.github/workflows/test.yml

```diff
@@ -0,0 +1,24 @@
+name: Run lint & unit tests
+
+on:
+  push:
+    branches: [ "main" ]
+  pull_request:
+    branches: [ "main" ]
+
+jobs:
+  build:
+    name: "Run unit tests"
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v3
+
+    - name: Install dependencies
+      run: pip install -r requirements.txt
+
+    - name: Run lint
+      run: make lint
+
+    - name: Run unit tests
+      run: make test
```

Comment:
Every file in this PR ends with a warning that the file does not contain a newline at the end. Change your IDE settings to add that in. Standardizing on either "always have a newline at the end" or "never have a newline at the end" cuts down on spurious diffs when different people work on the repo (and keep adding/removing them), and then the one to standardize on is "always have the newline" because that's the POSIX standard.

Not sure which IDE you use, but there might be tips on how to change it at the bottom of https://thoughtbot.com/blog/no-newline-at-end-of-file